ipatests/test_integration/test_adtrust_install.py

```diff
@@ -0,0 +1,190 @@
+        a new agent.
+        The Schema Compat plugin must be automatically configured on the
+        replica.
+        """
+        self.unconfigure_replica_as_agent(self.replicas[0])
+        cmd_input = (
+            # admin password:
+            self.master.config.admin_password + '\n' +
+            # WARNING: The smb.conf already exists. Running ipa-adtrust-install
+            # will break your existing samba configuration.
+            # Do you wish to continue? [no]:
+            'yes\n'
+            # Enable trusted domains support in slapi-nis? [no]:
+            '\n' +
+            # WARNING: 1 IPA masters are not yet able to serve information
+            # about users from trusted forests.
+            # Installer can add them to the list of IPA masters allowed to
+            # access information about trusts.
+            # If you choose to do so, you also need to restart LDAP service on
+            # those masters.
+            # Refer to ipa-adtrust-install(1) man page for details.
+            # IPA master[replica1.testrelm.test]?[no]:
+            'yes\n'
+        )
+        expected_re = '"ipactl restart".+"systemctl restart sssd"'
+        res = self.master.run_command(['ipa-adtrust-install', '--add-agents'],
+                                      stdin_text=cmd_input)
+        # The replica must have been restarted automatically, no msg required
+        assert not re.search(expected_re, res.stdout_text, re.DOTALL)
+
+    def test_add_agent_on_running_replica_with_compat(self):
+        """ Check ipa-addtrust-install --add-agents when the replica is running
+
+        Scenario: replica up and running
+        Call ipa-adtrust-install --add-agents --enable-compat and configure
+        the replica as a new agent.
+        The Schema Compat plugin must be automatically configured on the
+        replica.
+        """
+        self.unconfigure_replica_as_agent(self.replicas[0])
+
+        cmd_input = (
+            # admin password:
+            self.master.config.admin_password + '\n' +
+            # WARNING: The smb.conf already exists. Running ipa-adtrust-install
+            # will break your existing samba configuration.
+            # Do you wish to continue? [no]:
+            'yes\n'
+            # Enable trusted domains support in slapi-nis? [no]:
+            'yes\n' +
+            # WARNING: 1 IPA masters are not yet able to serve information
+            # about users from trusted forests.
+            # Installer can add them to the list of IPA masters allowed to
+            # access information about trusts.
+            # If you choose to do so, you also need to restart LDAP service on
+            # those masters.
+            # Refer to ipa-adtrust-install(1) man page for details.
+            # IPA master[replica1.testrelm.test]?[no]:
+            'yes\n'
+        )
+        expected_re = '"ipactl restart".+"systemctl restart sssd"'
```

Comment:
I suggest to reduce search string to "ipactl restart", otherwise if we have regression of https://pagure.io/freeipa/issue/8148 then this test will always be green.

(Note: if search string is simplified, re.search can be replaced with in)

Reply:
OK, good idea.